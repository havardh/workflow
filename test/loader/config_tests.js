// @flow
/* eslint-env jest */
/* eslint-disable no-unused-expressions, import/no-extraneous-dependencies */
import os from 'os';
import { sandbox } from 'sinon';

import RequireWrapper from '../../util/require';
import load from '../../loader/config';

const sinon = sandbox.create();

describe('load(configFile)', () => {
  const expectedFile = { file: 'content' };

  beforeEach(() => {
    sinon.stub(RequireWrapper, 'require');
    sinon.stub(process, 'cwd');
    sinon.stub(os, 'homedir');
    sinon.stub(console, 'log');

    // $FlowTodo
    RequireWrapper.require.throws(new Error());

    os.homedir.returns('/home/user');
    process.cwd.returns('/home/user/dev/workflow');
  });

  afterEach(() => {
    sinon.restore();
  });

  it('should require file directly', () => {
    RequireWrapper.require
      // $FlowTodo
      .withArgs('/path/to/file.js')
      .returns(expectedFile);

    const file = load('/path/to/file.js');

    expect(file).toEqual(expectedFile);
  });

  it('should require file from ~/.workflow', () => {
    RequireWrapper.require
      // $FlowTodo
      .withArgs('/home/user/.workflow/file.js')
      .returns(expectedFile);

    const file = load('file.js');

    expect(file).toEqual(expectedFile);
  });

  it('should require file from workflow examples', () => {
    RequireWrapper.require
      // $FlowTodo
      .withArgs('/home/user/dev/workflow/examples/file.js')
      .returns(expectedFile);

    const file = load('file.js');

    expect(file).toEqual(expectedFile);
  });

  it('should require in specific order', () => {
    RequireWrapper.require
      // $FlowTodo
      .withArgs('/home/user/dev/workflow/examples/file.js')
      .returns(expectedFile);

    load('file.js');

    // $FlowTodo
    const calls = RequireWrapper.require.getCalls();

    expect(calls[0].args[0]).toEqual('file.js');
    expect(calls[1].args[0]).toEqual('/home/user/.workflow/file.js');
    expect(calls[2].args[0]).toEqual('/home/user/dev/workflow/examples/file.js');
  });
});
