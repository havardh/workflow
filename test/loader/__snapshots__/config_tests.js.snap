// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load(configFile) should throw error when all loaders fails 1`] = `"Could not load file.js"`;
